File: reviewed-code/reviewed.rb
  class ShopifyAPIClient
  attr_accessor :shop_id
  SECRET_API_KEY = "asdasdasQWRESFSDFVSDFASDFSADFASDF123123ASDASD$%$%"
# I changed from attr_reader to attr_acessor, taking off the necessity of creating a setter, 
# and I created the initialize method to prevent that the variable shop_id could be used without any value.
  def initialize (shop_id) {
    @shop_id = shop_id
  }
# Instead of creating a hash with the username and password every time I would need it, I preferred to create this method that has the values that I need.
  def basic_auth
    {
      username: SECRET_API_KEY, 
      password: ''
    }
  end
# Instead of creating a lot of methods, I choose creating just these two methods that summarize well the job that those previous methods would do.
  def find(id)
    params = { basic_auth: basic_auth }
    json = some_http_library.get("https://www.shopify.this.is.a.sample.com/#{shop_id}/#{self.class.name.downcase}/#{id}", params)
    return JSON.parse(json)
  end

  def find_all
    params = { basic_auth: basic_auth }
    json = some_http_library.get("https://www.shopify.this.is.a.sample.com/#{shop_id}/#{self.class.name.downcase}", params)
    return JSON.parse(json)
  end
end
# Created two subclasses, one for orders, other for products, they will have find and find_all methods. I allowed order to have find with id, 
#because I assumed that there were more than one order, since the method name was "orders" not "order".
class Order < ShopifyAPIClient
# specific implementation
end

class Product < ShopifyAPIClient
# specific implementation
end

File: reviewed-code/reviewed.js
//Put IIFE to protect global environment and use strict.
(function(window, document) {
  'use strict';
//Created a function that it is a IIFE that has a lot of methods that have the same context, in this case, the context is making the order details appears on the HTML.
//The other reason to create a IIFE to store those methods is the fact that I can make methods public or private.
  var shopify = (function () {
//So I created shopify.writeOrderDetails to separate better each responsibility, it basically creates a new array, fill it strings that represent what I want to write on my html,
//after this, I use join on this array, passing a '\n' to the argument and write the return of this join. I used that way to make the code more readable and more organized.
    var writeOrderDetails = function (order, payment_info) {
      var finalHtml = [];
      finalHtml.push("<h1> Order receipt details </h1>");
      finalHtml.push("<p>Your order of" + order.products.name + "has been received</p>");
      finalHtml.push("<p>" + payment_info + "</p>");
      document.write(finalHtml.join('\n'));
    }
//On getPaymentInfo, I made two changes:I changed the name of the variable "p" to "payment_info", giving it a meaning,
//and I initialized it with the beginning of the text "Payment info:", instead of repeating it on all if's, and I changed a little bit of the logic, 
//instead of creating an unordered if/else statement, I made the logic follow two steps: 
//The first step is to see if the order is free or not, if yes, it just concatenate the order payment info and finish the if/else.
//If isn't free, it enter on the second step, that is to see which is the payment type,
//I used switch instead of an amount of if/else's to make it more organized.
  var getPaymentInfo = function (order) {
    var payment_info = 'Payment info: ';
    if (order.payment_type === 'free') {
      payment_info += order.payment.default_payment_info;
    } else {
      switch(order.payment_type) {
        case 'creditcard':
          payment_info += order.payment.getCardType + " " + order.payment.card_number;
          break;
        case 'paypal':
          payment_info += order.payment.paypal_info;
          break;
        case 'manual':
          payment_info += order.payment.manual_payment_info;
          break;
        default:
          payment_info += order.payment.default_payment_info;
        }
      return payment_info + "<p> was charged " + order.amount_in_dollars + "$" + "</p>";
      }
    }
//shopify.receipt will call writeOrderDetails and this call will receive the paymentInfo and the order.
    var receipt = function (order) {
      writeOrderDetails(order, getPaymentInfo(order));
    }
//I let receipt public, since it is the only method I really need to use outside this IIFE.
    return {
      receipt: receipt
    }
  })();
//Call receipt to see all order details on the html.
  shopify.receipt(order);

})(window, document);
